Tests/Reqnroll.GeneratorTests/VirtualFileSystem.cs

```diff
@@ -214,5 +214,13 @@ static void CheckSearchPattern(string searchPattern)
                 throw new ArgumentException("Illegal characters in path", "searchPattern");
             }
         }
+
+        public void CreateDirectory(string path)
+        {
+            if (!DirectoryExists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
```

Comment:
This is subtly a race condition and unnecessary: the directory could be created by another process between these two calls, _but_ the documentation for `Directory.CreateDirectory` indicates that it
> Creates all directories and subdirectories in the specified path unless they already exist.

So you don't need the check at all. 
```suggestion
            Directory.CreateDirectory(path);
```

My general advice is to always try to do file-system interaction as an atomic operation and look to handle error cases via exception or `Try-` methods.